tasks/depcheck.js

```diff
@@ -16,24 +16,32 @@ module.exports = function (grunt) {
   grunt.registerMultiTask('depcheck', 'Depcheck Grunt plugin', function () {
     var options = this.options({
       'withoutDev': false,
+      'failOnUnusedDeps': false,
       'ignoreDirs': ['.git','.svn','.hg','.idea','node_modules','bower_components'],
       'ingoreMatches': []
     });
 
     var dirsChecked = 0;
     var done = this.async();
+    var fail = false;
     var files = this.filesSrc;
 
     files.forEach(function (f) {
-      depcheck.check(path.resolve(f), options, function (unused) {
+      depcheck.check(path.resolve(f), {
+        ignoreDirs: options.ignoreDirs,
+        ingoreMatches: options.ingoreMatches,
+        withoutDev: options.withoutDev
```

Comment:
Hi @lijunle, thanks for having a look at this pull request. Initially I passed all options to `depcheck.check` directly, but then I thought depcheck might get confused because the new option `failOnUnusedDeps` is not part of the depcheck options and if it ever does become an option the code might not work anymore. However, if you think it's a sign of overengineering I can remove it again. :-)